.ci/pipelines/pr-gke.Jenkinsfile

```diff
@@ -20,65 +20,124 @@ pipeline {
+                            }
                         }
-                    }
-                }
-                stage("Run smoke E2E tests") {
-                    when {
-                        expression {
-                            notOnlyDocs()
+                        stage('check-license-header') {
+                            steps {
+                                sh 'make -C .ci TARGET=check-license-header ci'
+                            }
+                        }
+                        stage('check-predicates') {
+                            steps {
+                                sh 'make -C .ci TARGET=check-predicates ci'
+                            }
+                        }
+                        stage('shellcheck') {
+                            steps {
+                                sh 'make -C .ci TARGET=shellcheck ci'
+                            }
+                        }
+                        stage("reattach-pv-tool") {
+                            steps {
+                                sh 'make -C .ci TARGET=reattach-pv ci'
+                            }
+                        }
+                        stage("unit-tests") {
+                            steps {
+                                script {
+                                    env.SHELL_EXIT_CODE = sh(returnStatus: true, script: 'make -C .ci TARGET=unit-xml ci')
+                                    junit "unit-tests.xml"
+                                    sh 'exit $SHELL_EXIT_CODE'
+                                }
+                            }
+                        }
+                        stage("integration-tests") {
+                            steps {
+                                script {
+                                    env.SHELL_EXIT_CODE = sh(returnStatus: true, script: 'make -C .ci TARGET=integration-xml ci')
+                                    junit "integration-tests.xml"
+                                    sh 'exit $SHELL_EXIT_CODE'
+                                }
+                            }
                         }
                     }
-                    steps {
-                        sh '.ci/setenvconfig pr'
-                        script {
-                            env.SHELL_EXIT_CODE = sh(returnStatus: true, script: 'make -C .ci get-test-artifacts TARGET=ci-build-operator-e2e-run ci')
-
-                            sh 'make -C .ci TARGET=e2e-generate-xml ci'
-                            junit "e2e-tests.xml"
-
-                            sh 'exit $SHELL_EXIT_CODE'
+                }
+                stage('e2e-tests') {
+                    stages {
+                        stage('e2e-tests') {
+                            steps {
+                                script {
+                                    env.SHELL_EXIT_CODE = sh(returnStatus: true, script: 'make -C .ci TARGET="set-kubeconfig e2e-run" ci')
```

Comment:
`set-kubeconfig` does not deployer automatically set up the kubeconfig after bootstrapping the k8s cluster?

Reply:
Yes, the deployer does it well but it is not persisted outside the container, it is written in the default location `~/.kube/config`. It worked in the past because, we stayed in the same container to bootstrap the k8s cluster and then run the e2e tests, which is no more the case now.